### PR TITLE
Fixes #2558. MenuBar positions wrong in some situations.

### LINKED ISSUE #2558: MenuBar positions wrong in some situations
**Describe the bug**

Running FileDialog with the following code results in a MenuBar that is mis positioned and has trailing corruption on right.  I think it is to do with running using `Application.Run` directly without having a `Top` yet?  The issue does not manifest in the `FileDialogExamples` scenario.

```csharp
Application.Init ();
var dlg = new SaveDialog ("Save my file",
new List<IAllowedType> { new AllowedType("Comma Separated File",".csv") });

Application.Run (dlg);

if (!dlg.Canceled) {
    MessageBox.Query ("File Chosen", $"You chose {dlg.Path}", "Ok");
}

Application.Shutdown ();
```

image
_Opening the MenuBar when running with the above code_

Notice how the issue does not manifest in Scenario:

image
_Opening the MenuBar when running in Scenario_

### PR DESCRIPTION
**Pull request by BDisp**
menu-on-savedialog
Fixes #2558 - `Application.Top` added by the` Application.Init` must always be added to the toplevels stack and initialized. If a toplevel is a `IsOverlappedContainer` then the Application.Top must be set to the `OverlappedTop`.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by BDisp** [comment]
@tig I'm tying to re-enabling some unit tests that are commented. One is the `Dialog_In_Window_Without_Size_One_Button_Aligns` where you have this comment:
// BUGBUG: This seems wrong; is it a bug in Dim.Percent(85)??
and this output:
```
┌┌───────────────┐─┐
││               │ │
││     [ Ok ]    │ │
│└───────────────┘ │
└──────────────────┘
```
As you can see the height of the `Window` is 5. The available height inside is 5-2=3. The proportion is 3/5=0.6x100=60%. This is the ideal percentage for a `Dialog`, 60% and not 85%. In this case the new dimension is 5x60%=3. If you use the current value the dimension will be 5x85%=4.25, almost 5 which is the same size of the `Window`. So, I propose to change the `Dialog` percentage to 60%, at least for the height and the result is this:

```
┌──────────────────┐
│   ┌──────────┐   │
│   │  [ Ok ]  │   │
│   └──────────┘   │
└──────────────────┘
```

**Comment by BDisp** [comment]
@tig we can use the `Math.Ceiling` on `Percent` to round up or create an option to that.

```cs
internal override int Anchor (int width)
	{
		return (int)Math.Ceiling (width * factor);
	}
```
See the differences for height of 7:
Expected:
```
┌──────────────────┐
│   ┌──────────┐   │
│   │          │   │
│   │  [ Ok ]  │   │
│   └──────────┘   │
│                  │
└──────────────────┘
But Was:
┌──────────────────┐
│   ┌──────────┐   │
│   │          │   │
│   │          │   │
│   │  [ Ok ]  │   │
│   └──────────┘   │
└──────────────────┘

```

File: Terminal.Gui/Views/Toplevel.cs [hunk 1]
```diff
@@ -368,6 +368,72 @@ public static Toplevel Create ()
 		/// </summary>
 		public bool IsLoaded { get; private set; }
 
+		/// <inheritdoc/>
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Toplevel.cs:371]: This also will make it harder for us to get rid of `Toplevel` (#2502), and allow any view to act in an overlapping manner.

**Review by tig — CHANGES_REQUESTED** [review]
I believe this fix takes `Menubar` in the wrong direction: It makes it an even more "special" view. IMO, `MenuBar` (and `StatusBar`) has too many special cases baked into `Application` and `Toplevel` that should all be removed. It should be possible to use `Menubar` in any View and it should be possible for devs to build other Views that can be used instead of `Menubar` (or `StatusBar`).

**Pull request closed without merging**